L1Trigger/L1TTrackMatch/plugins/L1TrackJetEmulationProducer.cc

```diff
@@ -0,0 +1,671 @@
+    : trackToken_(
+          consumes<vector<TTTrack<Ref_Phase2TrackerDigi_>>>(iConfig.getParameter<InputTag>("L1TrackInputTag"))) {
+  trkZMax_ = (float)iConfig.getParameter<double>("trk_zMax");
+  trkPtMax_ = (float)iConfig.getParameter<double>("trk_ptMax");
+  trkPtMin_ = (float)iConfig.getParameter<double>("trk_ptMin");
+  trkEtaMax_ = (float)iConfig.getParameter<double>("trk_etaMax");
+  trkChi2dofMax_ = (float)iConfig.getParameter<double>("trk_chi2dofMax");
+  trkBendChi2Max_ = (float)iConfig.getParameter<double>("trk_bendChi2Max");
+  trkNPSStubMin_ = (int)iConfig.getParameter<int>("trk_nPSStubMin");
+  minTrkJetpT_ = iConfig.getParameter<double>("minTrkJetpT");
+  etaBins_ = (int)iConfig.getParameter<int>("etaBins");
+  phiBins_ = (int)iConfig.getParameter<int>("phiBins");
+  zBins_ = (int)iConfig.getParameter<int>("zBins");
+  d0CutNStubs4_ = (float)iConfig.getParameter<double>("d0_cutNStubs4");
+  d0CutNStubs5_ = (float)iConfig.getParameter<double>("d0_cutNStubs5");
+  lowpTJetMinTrackMultiplicity_ = (int)iConfig.getParameter<int>("lowpTJetMinTrackMultiplicity");
+  highpTJetMinTrackMultiplicity_ = (int)iConfig.getParameter<int>("highpTJetMinTrackMultiplicity");
+  displaced_ = iConfig.getParameter<bool>("displaced");
+  nStubs4DisplacedChi2Loose_ = (float)iConfig.getParameter<double>("nStubs4DisplacedChi2_Loose");
+  nStubs5DisplacedChi2Loose_ = (float)iConfig.getParameter<double>("nStubs5DisplacedChi2_Loose");
+  nStubs4DisplacedBendLoose_ = (float)iConfig.getParameter<double>("nStubs4Displacedbend_Loose");
+  nStubs5DisplacedBendLoose_ = (float)iConfig.getParameter<double>("nStubs5Displacedbend_Loose");
+  nStubs4DisplacedChi2Tight_ = (float)iConfig.getParameter<double>("nStubs4DisplacedChi2_Tight");
+  nStubs5DisplacedChi2Tight_ = (float)iConfig.getParameter<double>("nStubs5DisplacedChi2_Tight");
+  nStubs4DisplacedBendTight_ = (float)iConfig.getParameter<double>("nStubs4Displacedbend_Tight");
+  nStubs5DisplacedBendTight_ = (float)iConfig.getParameter<double>("nStubs5Displacedbend_Tight");
+
+  zStep_ = l1t::Scales::makeZ0(2.0 * trkZMax_ / zBins_);
+  etaStep_ = l1t::Scales::makeGlbEta(2.0 * trkEtaMax_ / etaBins_);  //etaStep is the width of an etabin
+  phiStep_ = l1t::Scales::makePhi(2.0 * M_PI / phiBins_);           ////phiStep is the width of a phibin
+
+  if (displaced_)
+    produces<l1t::TkJetWordCollection>("L1TrackJetsExtended");
+  else
+    produces<l1t::TkJetWordCollection>("L1TrackJets");
+}
+
+L1TrackJetEmulationProducer::~L1TrackJetEmulationProducer() {}
+
+void L1TrackJetEmulationProducer::produce(Event &iEvent, const EventSetup &iSetup) {
+  unique_ptr<l1t::TkJetWordCollection> L1L1TrackJetProducer(new l1t::TkJetWordCollection);
+
+  // For TTStubs
+  ESHandle<TrackerTopology> tTopoHandle;
+  iSetup.get<TrackerTopologyRcd>().get(tTopoHandle);
+  ESHandle<TrackerGeometry> tGeomHandle;
+  iSetup.get<TrackerDigiGeometryRecord>().get(tGeomHandle);
```

Comment:
Can you please switch to ESGetToken instead of ESHandle? (example in https://github.com/cms-l1t-offline/cmssw/pull/914/commits/a5fbdfef8ee7cc471a2797833bda6d89eeafa5a0)

Reply:
It's changed now.